subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/AbstractIntegrationSpec.groovy

```diff
@@ -634,4 +634,17 @@ tmpdir is currently ${System.getProperty("java.io.tmpdir")}""")
         this.ignoreCleanupAssertions = false
         recreateExecuter()
     }
+
+    /**
+     * This method exists as a workaround to avoid Spock printing power assertions when a comparison fails.
+     * <p>
+     * Any <strong>methods</strong> called in this block which do <strong>not</strong> make use of the {@code assert} statement
+     * will not cause a power assertion to be thrown upon failure.
+     * @see <a href="https://gist.github.com/tresat/716208856bbe01ed827c2aec5a847266">some examples</a>
+     *
+     * @param cls the code to be exercised containing methods which throw exceptions upon failure
+     */
+    void noPowerAssert(Closure<Void> cls) {
+        cls.call();
+    }
```

Comment:
IIUC, we would use this in a situation like this:
```
...
expect:
objectWithVerboseToString.someMethod == "someValue"
```
And we'd turn that into:
```
...
expect:
noPowerAssert {
   assert objectWithVerboseToString.someMethod == "someValue"
}
```

This doesn't seem to work completely in all cases.  e.g., if I wrap the call to `result.groupedOutput.task(":outgoingVariants").assertOutputContains` in `noPowerAssert`, it ends up including the body of the closure in the exception.

I think I wouldn't add this for now for a few reasons:
- It causes the IDE to complain because it thinks you're returning something from the closure sometimes. This happens when you call a method that returns something other than void inside the closure.
- It's easy to miss the call to `assert` and think we're asserting something when we're not
- It's probably better to try to make the power assert useful looking vs suppressing it

Reply:
We discussed this, and agreed to remove the method since it is less useful than hoped.